roxie/ccd/ccdcontext.cpp

```diff
@@ -3666,7 +3674,15 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
     }
     IUserDescriptor *queryUserDescriptor()
     {
-        return NULL; // TBD - Richard, where do user credentials for a roxie query come from
+        if (workUnit)
```

Comment:
This code is repeated in many places, seems like it could be commoned up somewhere.